### PR TITLE
MODOAIPMH-515 - Export harvest logs into csv format

### PR DESCRIPTION
**Pull request by obozhko-folio**
[MODOAIPMH-515](https://issues.folio.org/browse/MODOAIPMH-515) - Export harvest logs into csv format

## Purpose
 During the harvest, some errors are appeared due to the bad data (control characters, special symbols, not existing id etc.), but they are accessible only from the server logs. The goal of this task is to make such error records available to the user.

## Approach
Save every next error into DB and store the file into S3 at the end of the harvest. Link to S3 allows the user accessing the error logs and bad data right after the harvest is done.

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] Check logging
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: descriptors/ModuleDescriptor-template.json [hunk 1]
```diff
@@ -346,7 +346,13 @@
       { "name": "DB_DATABASE", "value": "okapi_modules" },
       { "name": "DB_QUERYTIMEOUT", "value": "2700000" },
       { "name": "DB_CHARSET", "value": "UTF-8" },
-      { "name": "DB_MAXPOOLSIZE", "value": "20" }
+      { "name": "DB_MAXPOOLSIZE", "value": "20" },
+      { "name": "S3_URL", "value": "http://127.0.0.1:9000/"},
+      { "name": "S3_REGION", "value": ""},
+      { "name": "S3_BUCKET", "value": "test-bucket2"},
+      { "name": "S3_ACCESS_KEY_ID", "value": "AKIAIOSFODNN7EXAMPLE"},
+      { "name": "S3_SECRET_ACCESS_KEY", "value": "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"},
```
Comment (alekGbuz) [inline descriptors/ModuleDescriptor-template.json:354]: @obozhko-folio , please remove any credentials

**Review by alekGbuz — APPROVED** [review]

File: src/main/java/org/folio/oaipmh/helpers/configuration/ErrorServiceConfig.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+  @Value("${minio.secretKey}")
+  private String secretKey;
+
+  @Value("#{ T(Boolean).parseBoolean('${minio.awsSdk}')}")
+  private boolean awsSdk;
+
+  @Bean
+  public FolioS3Client folioS3Client() {
+    log.info("Folio S3 client for error storage: endpoint {}, region {}, bucket {}, accessKey {}, secretKey {}, awsSdk {}",
```
Comment (firdavs-m) [inline src/main/java/org/folio/oaipmh/helpers/configuration/ErrorServiceConfig.java:36]: I guess it is not good idea to log secrets

File: src/main/java/org/folio/oaipmh/service/ErrorsService.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package org.folio.oaipmh.service;
+
+import io.vertx.core.Future;
+import org.folio.rest.jooq.tables.pojos.RequestMetadataLb;
+
+public interface ErrorsService {
+
+  void logLocally(String tenantId, String requestId, String instanceId, String errorMsg);
```
[the remark this answers is not in the document]
Comment (obozhko-folio) [reply src/main/java/org/folio/oaipmh/service/ErrorsService.java]: Renamed.

File: descriptors/ModuleDescriptor-template.json (hunk 1 above)
Comment (obozhko-folio) [reply descriptors/ModuleDescriptor-template.json]: Removed.

File: src/main/java/org/folio/oaipmh/helpers/configuration/ErrorServiceConfig.java (hunk 2 above)
Comment (obozhko-folio) [reply src/main/java/org/folio/oaipmh/helpers/configuration/ErrorServiceConfig.java:36]: Agree, secretKey was hidden.

**Review by khandramai — APPROVED** [review]

**Review by alekGbuz — APPROVED** [review]

**Review by siarhei-charniak — APPROVED** [review]

**Pull request merged**